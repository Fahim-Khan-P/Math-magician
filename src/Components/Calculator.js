import React, { useState } from 'react';
import calculate from '../Logic/calculate';

const Calculator = () => {
  const [obj, setObj] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const { total, next } = obj;
  const operationResult = !next ? total : next;

  const handleClick = (e) => {
    const result = calculate(obj, e.target.textContent);
    setObj(result);
  };

  return (
    <div className="claculatorView">
      <h2>Lets do some Math!</h2>
      <div className="calculator">
        <div className="displaySection">
          <input type="text" className="resultDisplay" readOnly value={operationResult || 0} />
        </div>

        <div className="buttonSection">
          <button type="button" className="button" onClick={handleClick}>AC</button>
          <button type="button" className="button" onClick={handleClick}>+/-</button>
          <button type="button" className="button" onClick={handleClick}>%</button>
          <button type="button" className="button operator" onClick={handleClick}>รท</button>

          <button type="button" className="button" onClick={handleClick}>7</button>
          <button type="button" className="button" onClick={handleClick}>8</button>
          <button type="button" className="button" onClick={handleClick}>9</button>
          <button type="button" className="button operator" onClick={handleClick}>x</button>

          <button type="button" className="button" onClick={handleClick}>4</button>
          <button type="button" className="button" onClick={handleClick}>5</button>
          <button type="button" className="button" onClick={handleClick}>6</button>
          <button type="button" className="button operator" onClick={handleClick}>-</button>

          <button type="button" className="button" onClick={handleClick}>1</button>
          <button type="button" className="button" onClick={handleClick}>2</button>
          <button type="button" className="button" onClick={handleClick}>3</button>
          <button type="button" className="button operator" onClick={handleClick}>+</button>

          <button type="button" className="button zeroBtn" onClick={handleClick}>0</button>
          <button type="button" className="button" onClick={handleClick}>.</button>
          <button type="button" className="button operator" onClick={handleClick}>=</button>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
