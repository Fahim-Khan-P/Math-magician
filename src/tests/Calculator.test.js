import React from 'react';
import renderer from 'react-test-renderer';
import { screen } from '@testing-library/react';
import Calculator from '../Components/Calculator';

test('calcul is available', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(screen.findByTitle('Lets do some math!')).toMatchSnapshot();
});
