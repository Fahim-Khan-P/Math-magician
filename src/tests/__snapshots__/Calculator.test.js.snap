// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calcul is available 1`] = `
<div
  className="claculatorView"
>
  <h2>
    Lets do some Math!
  </h2>
  <div
    className="calculator"
  >
    <div
      className="displaySection"
    >
      <input
        className="resultDisplay"
        readOnly={true}
        type="text"
        value={0}
      />
    </div>
    <div
      className="buttonSection"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="button operator"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="button operator"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="button operator"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="button operator"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="button zeroBtn"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="button operator"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`calcul is available 2`] = `Promise {}`;
